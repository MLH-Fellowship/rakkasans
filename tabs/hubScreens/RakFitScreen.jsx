import React, { useRef, useState, useEffect } from 'react';
import {
  View,
  Text,
  Image,
  StyleSheet,
  TouchableOpacity,
  Platform,
} from 'react-native';
import { useNavigation } from '@react-navigation/native';
import Carousel, { ParallaxImage } from 'react-native-snap-carousel';

import Card from '../../components/Card';
import Colors from '../../constants/Colors';
import Dimensions from '../../constants/Dimensions';
import SquareButton from '../../components/SquareButton';

const screenWidth = Dimensions.window.width;
const screenHeight = Dimensions.window.height;
const ENTRIES1 = [
  {
    title: 'City Jog',
    illustration: require('../../assets/images/fitness/fitness1.jpg'),
  },
  {
    title: 'Stretches',
    illustration: require('../../assets/images/fitness/fitness4.jpg'),
  },
  {
    title: 'Sprints',
    illustration: require('../../assets/images/fitness/fitness2.jpg'),
  },
  {
    title: 'Lifting',
    illustration: require('../../assets/images/fitness/fitness5.jpg'),
  },
  {
    title: 'Yoga',
    illustration: require('../../assets/images/fitness/fitness3.jpg'),
  },
];

export default function RakFitScreen() {
  const navigation = useNavigation();
  const [entries, setEntries] = useState([]);
  const carouselRef = useRef(null);
  useEffect(() => {
    setEntries(ENTRIES1);
  }, []);
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: Colors.white,
      width: '100%',
    },
    container2: {
      flex: 1,
      backgroundColor: Colors.white,
      justifyContent: 'center',
      alignItems: 'center',
      marginTop: 40,
    },
    buttonView: {
      width: '85%',
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      paddingBottom: 55,
    },
    challengetext: {
      fontSize: 25,
      color: Colors.primary,
    },
    fitnesstext: {
      fontSize: 22,
      color: Colors.primary,

      marginTop: 25,
      padding: 5,
    },
    item: {
      width: screenWidth - 60,
      height: screenHeight / 5.5,
    },
    scrollContainer: {
      flex: 1,
    },
    imageContainer: {
      flex: 1,
      marginBottom: Platform.select({ ios: 0, android: 1 }),
      backgroundColor: Colors.white,
      borderRadius: 30,
    },
    image: {
      ...StyleSheet.absoluteFillObject,
      resizeMode: 'cover',
    },
    title: {
      fontSize: 20,
      color: Colors.white,

      top: '-12%',
      left: '5%',
    },
  });
  const renderItem = ({ item, index }, parallaxProps) => (
    <View style={styles.item}>
      <ParallaxImage
        source={item.illustration}
        containerStyle={styles.imageContainer}
        style={styles.image}
        parallaxFactor={0.4}
        {...parallaxProps}
      />
      <Text style={styles.title}>{item.title}</Text>
    </View>
  );

  return (
    <View style={styles.container}>
      <View style={styles.container2}>
        <Image
          source={require('../../assets/images/Torri.png')}
          style={{ width: 30, height: 30, top: '-3%' }}
        />
        <Text style={styles.challengetext}>Rakkasan Challenge</Text>
        <TouchableOpacity>
          <Card>
            <Image
              source={require('../../assets/images/fitness/fitness2.jpg')}
              style={{
                width: screenWidth / 1.2,
                height: screenWidth / 2.1,
                borderRadius: 10,
              }}
            />
          </Card>
        </TouchableOpacity>

        <Text style={styles.fitnesstext}>Fast Fitness</Text>
        <View style={styles.scrollContainer}>
          <Carousel
            ref={carouselRef}
            sliderWidth={screenWidth}
            sliderHeight={screenWidth}
            itemWidth={screenWidth - 60}
            data={entries}
            renderItem={renderItem}
            hasParallaxImages
            firstItem={1}
          />
        </View>
        <View style={[styles.buttonView, { paddingTop: 10 }]}>
          <SquareButton
            name="food-apple"
            text="Nutrition"
            buttonSize={90}
            textSize={13}
            iconSize={50}
            // onPress={() => navigation.navigate("Program")}
          />
          <SquareButton
            name="food-fork-drink"
            text="DFAC"
            buttonSize={90}
            textSize={14}
            iconSize={50}
            // onPress={() => navigation.navigate("Program")}
          />
        </View>
      </View>
    </View>
  );
}
