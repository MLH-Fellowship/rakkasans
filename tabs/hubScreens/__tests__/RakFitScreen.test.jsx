import React from 'react';
import renderer from 'react-test-renderer';

import RakFitScreen from '../RakFitScreen';

jest.useFakeTimers();
jest.mock('@react-navigation/native');
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

it('renders correctly', () => {
  const tree = renderer.create(<RakFitScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
