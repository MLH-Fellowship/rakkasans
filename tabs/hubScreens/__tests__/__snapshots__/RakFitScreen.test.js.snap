// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 40,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 30,
          "top": "-3%",
          "width": 30,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#020a41",
          "fontFamily": "andale-mono",
          "fontSize": 25,
        }
      }
    >
      Rakkasan Challenge
    </Text>
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 15,
            "elevation": 5,
            "padding": 15,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.26,
            "shadowRadius": 8,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderRadius": 10,
              "height": 357.1428571428571,
              "width": 625,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#020a41",
          "fontFamily": "andale-mono",
          "fontSize": 22,
          "marginTop": 25,
          "padding": 5,
        }
      }
    >
      Fast Fitness
    </Text>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        activeAnimationOptions={null}
        activeAnimationType="timing"
        activeSlideAlignment="center"
        activeSlideOffset={20}
        apparitionDelay={0}
        automaticallyAdjustContentInsets={false}
        autoplay={false}
        autoplayDelay={1000}
        autoplayInterval={3000}
        callbackOffsetMargin={5}
        containerCustomStyle={Object {}}
        contentContainerCustomStyle={Object {}}
        contentContainerStyle={
          Array [
            Object {
              "paddingLeft": 30,
              "paddingRight": 30,
            },
            Object {},
          ]
        }
        data={Array []}
        decelerationRate="fast"
        directionalLockEnabled={true}
        disableVirtualization={false}
        enableMomentum={false}
        enableSnap={true}
        firstItem={1}
        getItem={[Function]}
        getItemCount={[Function]}
        hasParallaxImages={true}
        horizontal={true}
        inactiveSlideOpacity={0.7}
        inactiveSlideScale={0.9}
        inactiveSlideShift={0}
        initialNumToRender={7}
        invertStickyHeaders={false}
        inverted={false}
        itemWidth={690}
        keyExtractor={[Function]}
        layout="default"
        lockScrollTimeoutDuration={1000}
        lockScrollWhileSnapping={false}
        loop={false}
        loopClonesPerSide={3}
        maxToRenderPerBatch={15}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onResponderRelease={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onStartShouldSetResponderCapture={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        overScrollMode="never"
        pinchGestureEnabled={false}
        removeClippedSubviews={true}
        renderItem={[Function]}
        scrollEnabled={true}
        scrollEventThrottle={1}
        scrollsToTop={false}
        shouldOptimizeUpdates={true}
        showsHorizontalScrollIndicator={false}
        showsVerticalScrollIndicator={false}
        slideStyle={Object {}}
        sliderHeight={750}
        sliderWidth={750}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flexDirection": "row",
            "opacity": 0,
            "width": 750,
          }
        }
        swipeThreshold={20}
        updateCellsBatchingPeriod={50}
        useScrollView={false}
        vertical={false}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={15}
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-evenly",
            "paddingBottom": 55,
            "width": "85%",
          },
          Object {
            "paddingTop": 10,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 90,
                "width": 90,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "paddingBottom": 6,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "fontSize": 13,
                },
              ]
            }
          >
            Nutrition
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 15,
            "elevation": 5,
            "opacity": 1,
            "padding": 10,
            "shadowColor": "#a0a0a0",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.4,
            "shadowRadius": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Object {
                "height": 90,
                "width": 90,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "paddingBottom": 6,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "andale-mono",
                },
                Object {
                  "fontSize": 14,
                },
              ]
            }
          >
            DFAC
          </Text>
          <View
            style={
              Object {
                "alignItems": "center",
                "borderRadius": 25,
                "height": 50,
                "justifyContent": "center",
                "width": 50,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
